File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/util/SwipeRestrictViewPager.java
package com.igorganapolsky.vibratingwatchapp.util;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class SwipeRestrictViewPager extends ViewPager {

    private boolean isSwipeAvailable = false;

    public SwipeRestrictViewPager(@NonNull Context context) {
        super(context);
    }

    public SwipeRestrictViewPager(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return isSwipeAvailable && super.onTouchEvent(event);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        return isSwipeAvailable && super.onInterceptTouchEvent(event);
    }

    public void setIsSwipeAvailable(boolean isSwipeAvailable) {
        this.isSwipeAvailable = isSwipeAvailable;
    }

    public boolean isSwipeAvailable() {
        return isSwipeAvailable;
    }

}

File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/core/SwipeRestrictViewPager.kt
package com.igorganapolsky.vibratingwatchapp.core

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

/**
 * A ViewPager that controls swiping availability on timer creation screens.
 */
class SwipeRestrictViewPager(context: Context, attrs: AttributeSet?) : ViewPager(context, attrs) {
    var isSwipeAvailable = false

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        return isSwipeAvailable && super.onTouchEvent(event)
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        return isSwipeAvailable && super.onInterceptTouchEvent(event)
    }

}
